src/Swift.Bindings/src/Parser/SwiftABIParser.cs

```diff
@@ -278,9 +268,7 @@ private List<BaseDecl> CollectDeclarations(IEnumerable<Node> nodes, BaseDecl par
                     return null;
             }
 
-            typeRecord.IsProcessed = true;
-
-            if (node.Children != null && decl != null)
+            if (decl != null)
```

Comment:
favor `is not null` over `!= null` to avoid possible issues that arise from operator implementations.